extensions/scala/tests/src/main/scala/org/virtuslab/intellij/extensions/ScalaPluginBuilder.scala

```diff
@@ -39,26 +39,25 @@ object ScalaPluginBuilder extends DependencyBuilder(Id("scala")) {
 
   private def clone(repository: Git) = {
     val localRepo = Files.createTempDirectory("scala-plugin-repo")
-    val command: Array[String] = repository.branch match {
-      case None         => Array("git", "clone", repository.path.toString, localRepo.toString)
-      case Some(branch) => Array("git", "clone", repository.path.toString, branch, localRepo.toString)
+    val cloned = Shell.run("git", "clone", repository.path.toString, localRepo.toString)
+    if (cloned.exitCode != 0) throw new IllegalStateException(s"Could not clone git $repository")
+    repository.ref.foreach { ref =>
+      val checkout = Shell.run(in = localRepo, "git", "checkout", ref)
+      if (checkout.exitCode != 0) throw new IllegalStateException(s"Could not checkout $ref in $repository")
     }
```

Comment:
I'm not sure if that is any simpler. Is it because we can clone only branches?

Reply:
It is not meant to be simpler, it is meant to support checking out commits. We can only clone branches, yes, so it would not work with commit hashes.